### PR TITLE
Added MakeReservationFromDeck reaction

### PR DESCRIPTION
**Pull request by v4m3rrr**
Solved issue #64

### CODE REVIEW

File: src/main/java/com/github/splendor_mobile_game/websocket/handlers/reactions/MakeReservationFromDeck.java [hunk 1]
```diff
@@ -76,9 +89,178 @@ public MakeReservationFromDeck(int connectionHashCode, UserMessage userMessage,
...
+                new CardDataResponse(
+                    card.getUuid(), 
+                    card.getPoints(), 
+                    card.getAdditionalToken(), 
+                    new TokensDataResponse(
+                        card.getCost(TokenType.RUBY), 
+                        card.getCost(TokenType.EMERALD), 
+                        card.getCost(TokenType.SAPPHIRE), 
+                        card.getCost(TokenType.DIAMOND), 
+                        card.getCost(TokenType.ONYX)
```
Comment (kestivvi) [inline src/main/java/com/github/splendor_mobile_game/websocket/handlers/reactions/MakeReservationFromDeck.java:178]: I like this way of building a response.

File: src/main/java/com/github/splendor_mobile_game/websocket/handlers/reactions/MakeReservationFromDeck.java [hunk 2]
```diff
@@ -76,9 +89,178 @@ public MakeReservationFromDeck(int connectionHashCode, UserMessage userMessage,
...
+        }
+
+        //Check if user has not maxed tokens
+        if(user.getTokenCount()>=10){
+            throw new TokenCountException("Reached max token count on hand");
+        }
+
+        //Check if if gold tokens are available
+        if(game.getTokenCount(TokenType.GOLD_JOKER)<=0){
+            throw new TokenCountException("There is not enough golden joker on table");
```
Comment (kestivvi) [inline src/main/java/com/github/splendor_mobile_game/websocket/handlers/reactions/MakeReservationFromDeck.java]: User can reserve a card when there is no more gold tokens. In such situation they reserve a card, but don't get the gold token.

File: src/main/java/com/github/splendor_mobile_game/websocket/handlers/reactions/MakeReservationFromDeck.java [hunk 3]
```diff
@@ -76,9 +89,178 @@ public MakeReservationFromDeck(int connectionHashCode, UserMessage userMessage,
...
+        }
+
+        //Check throughout game reservation count
+        if(user.getThroughoutGameReservationCount()>=5){
+            throw new UserReservationException("User reached the limit of reserved cards per game");
+        }
+
+        //Check if user has not maxed tokens
+        if(user.getTokenCount()>=10){
+            throw new TokenCountException("Reached max token count on hand");
```
Comment (kestivvi) [inline src/main/java/com/github/splendor_mobile_game/websocket/handlers/reactions/MakeReservationFromDeck.java:241]: I don't remeber exact rules, but I think user can reserve a card in such situation, they simply just don't get the gold token. 

Do you remember what should be done in a such case? @Vertonowsky @KamilDevelopments @EaZZyHK @sssadpeach15

File: src/main/java/com/github/splendor_mobile_game/websocket/handlers/reactions/MakeReservationFromDeck.java [hunk 4]
```diff
@@ -76,9 +89,178 @@ public MakeReservationFromDeck(int connectionHashCode, UserMessage userMessage,
...
+
+
+
+
+
+
+
+
+
+
```
Comment (kestivvi) [inline src/main/java/com/github/splendor_mobile_game/websocket/handlers/reactions/MakeReservationFromDeck.java]: I don't let those blank lines of code to be approved. If you want to get more lines of code, to be higher in the ranking you should add java docs to every class, method and field you've added. Free lines man.

**Review by kestivvi — CHANGES_REQUESTED** [review]
Fascinating 😮😮 You've solved like 3 high priority not so easy issues!! That's a lot of code!

I put some comments down the road.

**Review by kestivvi — DISMISSED** [review]
Magnificent 💎🏆🪄🍕

Just resolve merge conflicts and it goes in!

File: src/main/java/com/github/splendor_mobile_game/websocket/handlers/reactions/MakeReservationFromDeck.java [hunk 5]
```diff
@@ -1,53 +1,67 @@
...
  * {
- *      "messageContextId": "02442d1b-2095-4aaa-9db1-0dae99d88e03",
- *      "type": "RESERVATION_FROM_TABLE_RESPONSE",
+ *      "contextId": "02442d1b-2095-4aaa-9db1-0dae99d88e03",
+ *      "type": "RESERVATION_FROM_DECK_ANNOUNCEMENT",
  *      "result": "OK",
  *      "data": {
+ *          "userUuid":"01901b0e-a78b-4a65-bbd3-0065948dc127",
  *          "card": {
  *              "id": "0ba9cba8-3bc0-42fe-b24f-25d7b52fcd2c",
```
Comment (kestivvi) [inline src/main/java/com/github/splendor_mobile_game/websocket/handlers/reactions/MakeReservationFromDeck.java]: There should be `uuid`

**Review by kestivvi — CHANGES_REQUESTED** [review]

**Review by kestivvi — APPROVED** [review]
🤩🤩🍕

**Review by Vertonowsky — APPROVED** [review]

**Pull request merged**